pkgs/tools/misc/fastfetch/default.nix

```diff
@@ -0,0 +1,132 @@
+{
+  fetchFromGitHub,
+  lib,
+  stdenv,
+  testers,
+  nix-update-script,
+  #nativeBuildInputs
+  cmake,
+  makeWrapper,
+  pkg-config,
+  #buildInputs
+  dbus,
+  dconf,
+  glib,
+  pciutils,
+  zlib,
+  #overrides
+  withFullDeps ? false,
+
+  enableChafa ? false,
+  chafa,
+
+  enableImageMagick ? false,
+  imagemagick_light,
+
+  enableNM ? false,
+  networkmanager,
+
+  enableOpenCL ? true,
+  opencl-headers,
+  ocl-icd,
+
+  enableOpenGL ? true,
+  libglvnd,
+
+  enablePulseAudio ? false,
+  libpulseaudio,
+
+  enableRpm ? false,
+  rpm,
+
+  enableSqlite ? false,
+  sqlite,
+
+  enableVulkan ? true,
+  vulkan-loader,
+
+  enableWayland ? true,
+  wayland,
+
+  enableX11 ? true,
+  libxcb,
+  libXrandr,
+
+  enableXfce ? false,
+  xfce,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+
+  pname = "fastfetch";
+  version = "1.12.2";
+
+  src = fetchFromGitHub {
+    owner = "LinusDierheimer";
+    repo = finalAttrs.pname;
+    rev = finalAttrs.version;
+    hash = "sha256-l9fIm7+dBsOqGoFUYtpYESAjDy3496rDTUDQjbNU4U0=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    makeWrapper
+    pkg-config
+  ];
+
+  runtimeDependencies =
+    [
+      dbus
+      dconf
+      glib
+      pciutils
+      zlib
+    ]
+    ++ lib.optional (withFullDeps || enableChafa) chafa
+    ++ lib.optional (withFullDeps || enableImageMagick) imagemagick_light
+    ++ lib.optional (withFullDeps || enableNM) networkmanager
```

Comment:
most of these are linux specific and should probably add a flag for linux dependencies